src/psyclone/psyGen.py

```diff
@@ -566,11 +566,12 @@ def unique_declarations(self, argument_type, access=None):
         :raises InternalError: if an invalid access is specified.
 
         '''
-        if argument_type not in VALID_ARG_TYPE_NAMES:
-            raise InternalError(
-                "Invoke.unique_declarations() called with an invalid argument "
-                "type. Expected one of {0} but found '{1}'".
-                format(str(VALID_ARG_TYPE_NAMES), argument_type))
+        for argtype in argument_types:
```

Comment:
any works on iterators so we shouldn't need the list? i.e. `if any(argtype not in VALID_ARG_TYPE_NAMES for argtype in argument_types):`

Reply:
Very elegant but unfortunately it only returns `True` or `False` but not the type of the offending argument nor how to find it, and that is useful for the error message.
Then I tried this:
```
        invalid_arg = [arg for arg in argument_types if arg not in VALID_ARG_TYPE_NAMES]
        if invalid_arg:
            raise InternalError(...
```
and that returns a list. I figured out this is actually correct behaviour and also practical. Now that we have a list of argument types instead of one there is a possibility that more than one will be invalid and it is better to list them all than check for one at a time. I modified the error messages and docstrings accordingly.

Reply:
@rupertford, what you wrote works but also does not return the names of invalid argument types for a useful error message. I decided to go for the list comprehension that returns the names of invalid types.

Reply:
I would have just printed the whole list in the event of an error but it matters not :-)

Reply:
Ah, I am with you now! Went for Rupert's comprehension and just printed the whole list.